ui-core/src/components/inputs/TolerancePicker.tsx

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useState, useRef, useMemo, useEffect } from 'react';
+import Input, { InputProps } from './Input';
+import Modal from '../Modal';
+import cx from 'classnames';
+import { StatusWithMessage } from './StatusMessage';
+
+type ToleranceRange = { label: string; value: number };
+
+export type ToleranceValues = {
+  minusTolerance: number;
+  plusTolerance: number;
+};
+
+export type TolerancePickerProps = Omit<InputProps, 'value'> & {
+  onToleranceChange: (toleranceValues: ToleranceValues) => void;
+  toleranceRanges: ToleranceRange[];
+  toleranceValues?: ToleranceValues;
+};
+
+const TolerancePicker: React.FC<TolerancePickerProps> = ({
+  onToleranceChange,
+  toleranceRanges,
+  toleranceValues: { minusTolerance, plusTolerance } = {
+    minusTolerance: toleranceRanges[0].value,
+    plusTolerance: toleranceRanges[0].value,
+  },
+  ...inputProps
+}) => {
+  const formatToleranceValue = (minusIndex: number, plusIndex: number) =>
+    `-${toleranceRanges[minusIndex]?.label || ''}/+${toleranceRanges[plusIndex]?.label || ''}`;
+
+  const [showPicker, setShowPicker] = useState(false);
+  const [inputValue, setInputValue] = useState(formatToleranceValue(0, 0));
+  const [warningStatus, setWarningStatus] = useState<StatusWithMessage | undefined>(undefined);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const { minusToleranceIndex, plusToleranceIndex } = useMemo(() => {
+    const minusToleranceIndex = toleranceRanges.findIndex(
+      (range) => range.value === minusTolerance
+    );
+    const plusToleranceIndex = toleranceRanges.findIndex((range) => range.value === plusTolerance);
+    if (minusToleranceIndex < 0 || plusToleranceIndex < 0)
+      setWarningStatus({
+        status: 'warning',
+        message: "One of the values isn't included in the ranges provided.",
+      });
+    return { minusToleranceIndex, plusToleranceIndex };
+  }, [minusTolerance, plusTolerance, toleranceRanges]);
+
+  useEffect(() => {
+    setInputValue(formatToleranceValue(minusToleranceIndex, plusToleranceIndex));
+  }, [minusTolerance, plusTolerance]);
+
+  return (
+    <div className="tolerance-picker">
+      <div>
+        <Input
+          {...inputProps}
+          value={inputValue}
+          // Event not used for now
+          onChange={(e) => {
+            setInputValue(e.target.value);
+            setShowPicker(true);
+          }}
+          statusWithMessage={warningStatus}
+          readOnly
+          onClick={() => setShowPicker(!showPicker)}
+          type="text"
+          ref={inputRef}
+        />
+      </div>
+
+      <Modal inputRef={inputRef} isOpen={showPicker} onClose={() => setShowPicker(false)}>
+        <div className="time-tolerance">
+          <div className="time-picker-section">
+            <div className="tolerance-grid">
+              {toleranceRanges.map((tolerance, index) => {
+                const { label, value } = tolerance;
```

Comment:
nit: You can even destructure the tolerance (but I'm not sure that it will improve the readibility though, as you wish)
```suggestion
              {toleranceRanges.map(({ label, value } , index) => (
```